internal/db/project.go

```diff
@@ -39,6 +40,29 @@ func DeleteProject(ctx context.Context, tx pgx.Tx, project int64) error {
 	return nil
 }
 
+func ListProjects(ctx context.Context, tx pgx.Tx) ([]*pb.Project, error) {
+	rows, err := tx.Query(ctx, `
+		SELECT id, latest_version
+		FROM dl.projects
+	`)
+	if err != nil {
+		return nil, fmt.Errorf("snapshotProjects query: %w", err)
+	}
+
+	projects := []*pb.Project{}
+
+	for rows.Next() {
```

Comment:
Yeah this is why `rows.Close()` rarely leaves dangling query objects around, because we typically iterate over all of the results quite quickly.

That being said it is probably still good practice to add `defer rows.Close` after most queries in case we throw an error while scanning the rows.